dvc/repo/brancher.py

```diff
@@ -53,6 +59,9 @@ def brancher(  # noqa: E302
                 # ignore revs that don't contain repo root
                 # (i.e. revs from before a subdir=True repo was init'ed)
                 if self.tree.exists(self.root_dir):
-                    yield ", ".join(names)
+                    if sha_only:
+                        yield sha
+                    else:
+                        yield ", ".join(names)
```

Comment:
When using names we get a comma-separated list of `hexsha, branch1, branch2, ...`, but rather than re-splitting/stripping/etc the yielded list later in `exp show` it seemed cleaner to just handle the case where we don't need the full name list directly in brancher.